server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -667,6 +669,65 @@ def ifthenelse(
 
         return StepContext(query, columns + [step.new_column])
 
+    @staticmethod
+    def _get_join_type(join_type: Literal['left', 'inner', 'left outer']) -> JoinType:
+        match join_type:
+            case 'left':
+                return JoinType.left
+            case 'left outer':
+                return JoinType.left_outer
+            case 'inner':
+                return JoinType.inner
+
+    def join(
```

Comment:
good stuff! 
I see one thing that I guess we wont be able to solve: after a custom step, we'll lose the list of columns.
Thus I guess join after a custom step won't work.

Reply:
Hmm I think it will work, it's just that our output columns will be weird (something like `["a", "b", "*"]`)... I'll add a fixture!

Reply:
done in https://github.com/ToucanToco/weaverbird/pull/1407/commits/096b61ab1996a3831010b8c3b7015f7dcf5bff8d